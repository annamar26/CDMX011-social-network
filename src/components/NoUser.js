import { onNavigate } from "../routes.js";

const noUser = () => {
  const conteiner = document.createElement('div');

  const html = `<nav>
      <img class='logo' src='./icons&img/logo.png'>
     
      
      </nav>
      <div class='noUser-view signUp-container'>
      <h2>Debes iniciar sesion para ver el contenido de la pagina</h2>
      <br>
      <br>
      <input id='loginSubmit' class='button' type=button value='Inicia SesiÃ³n'>
      </div>`;

  conteiner.innerHTML = html;
  const volver = conteiner.querySelector('#loginSubmit');
  volver.addEventListener('click', ()=>{
    onNavigate('/login');
  });

  return conteiner;
};

export default noUser;
