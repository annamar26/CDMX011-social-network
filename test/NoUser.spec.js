/**
 * @jest-environment jsdom
 */
import NoUser from '../src/components/NoUser';
import { render } from '../src/utils.js';

describe('NoUser', () => {
  document.body.innerHTML = '<div id="root"></div>';
  const rootDiv = document.getElementById('root');
  it('should render NoUser page', () => {
    render(rootDiv, NoUser());

    expect(rootDiv).toMatchSnapshot();
  });
});
