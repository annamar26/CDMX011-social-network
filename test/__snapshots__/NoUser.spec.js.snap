// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoUser should render NoUser page 1`] = `
<div
  id="root"
>
  <div>
    <nav>
      
      
      <img
        class="logo"
        src="./icons&img/logo.png"
      />
      
     
      
      
    </nav>
    
      
    <div
      class="noUser-view signUp-container"
    >
      
      
      <h2>
        Debes iniciar sesion para ver el contenido de la pagina
      </h2>
      
      
      <br />
      
      
      <br />
      
      
      <input
        class="button"
        id="loginSubmit"
        type="button"
        value="Inicia SesiÃ³n"
      />
      
      
    </div>
  </div>
</div>
`;
